### PR TITLE
chore: [receiver/kafka] docs add example of connecting to kafka with sasl +tls

### PR DESCRIPTION
**Pull request by povilasv**
**Description:** 

Adding an example on how to connect to kafka that needs both tls and sasl config.

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31931

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: receiver/kafkareceiver/README.md [hunk 1]
```diff
@@ -133,3 +146,6 @@ we will get a log record in collector similar to:
 
 - Here you can see the kafka record header `header1` and `header2` being added to resource attribute.
 - Every **matching** kafka header key is prefixed with `kafka.header` string and attached to resource attributes.
+
+
+##
```
Comment (songy23) [inline receiver/kafkareceiver/README.md:151]: Remove?

**Review by songy23 — APPROVED** [review]

**Review by crobert-1 — APPROVED** [review]

File: receiver/kafkareceiver/README.md [hunk 2]
```diff
@@ -89,14 +89,27 @@ The following settings can be optionally configured:
...
+```yaml
+receivers:
+  kafka:
+    auth:
+      sasl:
+        username: "user"
+        password: "secret"
+        mechanism: "SCRAM-SHA-512"
+    tls:
+      insecure: false
```
Comment (evan-bradley) [inline receiver/kafkareceiver/README.md:111]:
```suggestion
      tls:
        insecure: false
```
I think this needs to be indented another line, the TLS config is under the `auth` config.

**Comment by evan-bradley** [comment]
@povilasv @MovieStoreGuy What do you think about this? I would like to stay consistent with the OTLP receiver, which does:

```yaml
receivers:
  otlp:
    protocols:
      grpc:
```

So we could just have it such that setting `tls` enables it. But we could also have a `tls_enabled` key, or try to put an `enabled` field in the `tls` section.

File: receiver/kafkareceiver/README.md [hunk 3]
```diff
@@ -89,14 +89,27 @@ The following settings can be optionally configured:
...
+```yaml
+receivers:
+  kafka:
+    auth:
+      sasl:
+        username: "user"
+        password: "secret"
+        mechanism: "SCRAM-SHA-512"
+      tls:
+        insecure: false
```
Comment (pavolloffay) [inline receiver/kafkareceiver/README.md:111]: The false is default so it can be omitted

**Comment by evan-bradley** [comment]
Sorry, I lost track of this.

>And it' seems that it renders tls section to nil and doesn't enable TLS.

My bad, I should have been a bit clearer: I think we can update the code to work more like the OTLP receiver to allow merely setting the `tls` key to mean "enable TLS" by creating a default config, then setting it to `nil` if the user doesn't specify the key. Here's how the OTLP receiver does each:

1. Creating a default config: https://github.com/open-telemetry/opentelemetry-collector/blob/main/receiver/otlpreceiver/factory.go#L41
2. The default config is kept if the user sets the key, otherwise set to `nil` and disabled if they do not: https://github.com/open-telemetry/opentelemetry-collector/blob/main/receiver/otlpreceiver/config.go#L68

**Comment by MovieStoreGuy** [comment]
> @povilasv @MovieStoreGuy What do you think about this? I would like to stay consistent with the OTLP receiver, which does:
> 
> ```yaml
> receivers:
>   otlp:
>     protocols:
>       grpc:
> ```
> 
> So we could just have it such that setting `tls` enables it. But we could also have a `tls_enabled` key, or try to put an `enabled` field in the `tls` section.

The config needs some love since it is somewhat of a sprawling mess as a side affect of how Kafka library exposes its configuration. I agree it needs some love, but I don't want to do a bunch of subtle changes that makes it hard to understand what changed.

**Review by MovieStoreGuy — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**